src/libstd/sys/mod.rs

```diff
@@ -40,8 +40,15 @@ cfg_if! {
         mod windows;
         pub use self::windows::*;
     } else if #[cfg(target_os = "cloudabi")] {
-        mod cloudabi;
-        pub use self::cloudabi::*;
+        cfg_if! {
+            if #[cfg(target_arch = "wasm32")] {
+                mod wasm_cloudabi;
+                pub use self::wasm_cloudabi::*;
```

Comment:
Out of interest, is it already necessary to break this down at this level? What prevents us from converging `cloudabi` <-> `wasm_cloudabi` even more?

Reply:
There are some dependencies that could make life harder (e.g. dependency on libc). It is experimental target, so the decision was made to disconnect it from wasm32 and cloudabi.

Reply:
> (e.g. dependency on libc)

Just want to mention that when I ported Rust to CloudABI, I kept a dependency on libc because it was the path of least resistance. I guess Rust had a stronger reliance on the C library at the time than it has now. I tried to already reduce the interaction between the two, but couldn't eliminate it entirely.

If merging `cloudabi` and `wasm_cloudabi` means that there is less of a reliance on the C library for non-wasm targets, that would be awesome in my opinion.